src/containers/CheckoutPage.tsx

```diff
@@ -613,40 +567,18 @@ class CheckoutPage extends React.Component<CheckoutPageProps, CheckoutPageState>
   }
 
   renderPaymentSelector() {
-    const { profileData, openNewPaymentModal } = this.state;
-    const isDisabled = !(!profileData || (profileData && profileData._emails[0]._element));
+    const { profileData } = this.state;
 
     return (
-      <div>
-        <h2>
-          {intl.get('payment-method')}
-        </h2>
-        <div data-region="paymentMethodSelectorsRegion" className="checkout-region-inner-container">
-          {this.renderPayments()}
-        </div>
-        <button className="ep-btn primary wide checkout-new-payment-btn" disabled={isDisabled} type="button" onClick={() => { this.newPayment(); }}>
-          {intl.get('add-new-payment-method')}
-        </button>
-        <Modal open={openNewPaymentModal} onClose={this.handleCloseNewPaymentModal}>
-          <div className="modal-lg new-payment-modal">
-            <div className="modal-content">
-              <div className="modal-header">
-                <h2 className="modal-title">
-                  {intl.get('new-payment-method')}
-                </h2>
-              </div>
-              <div className="modal-body">
-                {
-                  localStorage.getItem(`${Config.cortexApi.scope}_oAuthRole`) === 'PUBLIC' ? (
```

Comment:
@dusanradovanovic Need to place this anonymous user check through to `ProfilePaymentMethodsMain` component.  Currently, the payment form on the checkout page will always save the payment information to the profiles resource however saving to the profile is not available for anonymous users

Reply:
I would recommend doing the localstorage check on the `***page.tsx` file and passing the props down to the `paymentform.main.tsx` file that dictates which resource it will post payment information too.  This ensures that the components don't make an assumption as to how login information in stored.